synfig-studio/src/gui/widgets/widget_timeslider.cpp

```diff
@@ -230,7 +233,7 @@ Widget_Timeslider::on_draw(const Cairo::RefPtr<Cairo::Context> &cr)
 	const etl::handle<TimeModel> & time_model = time_plot_data->time_model;
 
 	// Draw the time line...
-	double tpx = time_plot_data->get_pixel_t_coord(time_plot_data->time) + 0.5;
+	double tpx = time_plot_data->get_pixel_t_coord(time_plot_data->time) + EXTRA_TIMETRACK_SPACE;
```

Comment:
Maybe `time_plot_data` should handle this extra space internally in its members `get_pixel_t_coord()` and `get_t_from_pixel_coord()`

Reply:
and `get_double_pixel_t_coord()`

Reply:
In my current implementation where I made the fix on `TimePlotData::recompute_geometry_data()`
managing **k** value in this function makes sense

If we had to let `get_pixel_t_coord()` take care of k then there is no point for recompute... function to exist because we'd have to
1. re-calculate k first (notice the get_width()-EXTRA_TIMETRACK_SPACE*2 before being divided by time difference)
2. and then return time coordinate

within the get_pixel_t_coord() function itself which kind of breaks the rule "let function take care of one thing only"

I feel ok with my current solution
The only drawback that new contributors implementing a time slider related widget take note of pushing their canvas drawing by 16px

But if you feel otherwise I understand
Maybe you can take over this PR and apply your creative fixes :)